src/components/capture/creation/SpecForm.tsx

```diff
@@ -29,45 +29,50 @@ const defaultAjv = createAjv({ useDefaults: true });
 function NewCaptureSpecForm(props: NewCaptureSpecFormProps) {
     const { state, dispatch, endpoint, readonly, displayValidation } = props;
 
+    const { isIdle, isLoading, isSuccess, error, data } =
+        useConnectorImageSpec(endpoint);
+
     const {
-        loading,
-        error,
-        data: {
-            specSchema,
-            links: { discovery, documentation },
-        },
-    } = useConnectorImageSpec(endpoint);
+        links: { discovery, documentation },
+        endpointSchema,
+    } = data;
 
     useEffect(() => {
-        dispatch({
-            payload: discovery,
-            type: ActionType.NEW_DISCOVERY_LINK,
-        });
+        if (discovery.length > 0) {
+            dispatch({
+                payload: discovery,
+                type: ActionType.NEW_DISCOVERY_LINK,
+            });
+        }
```

Comment:
Since we have a default state we need to make sure some real values were put in before dispatching.